File: python/cocina.py
from cocinero import Cocinero
from pedido import Pedido
import threading
import random

def main():
    random.seed()  # Inicializar generador aleatorio

    # Lista de platos disponibles
    platos_disponibles = [
        "Pizza Margarita", "Lasaña Boloñesa", "Sushi Variado", "Paella Mixta", "Tacos de Pollo",
        "Ensalada César", "Hamburguesa Doble", "Canelones de Espinacas", "Sopa de Tomate", "Pollo al Curry",
        "Burritos de Ternera", "Filete con Patatas", "Gazpacho Andaluz", "Pasta Carbonara", "Raviolis de Queso",
        "Huevos Rotos con Jamón", "Croquetas Caseras", "Pulpo a la Gallega", "Tarta de Queso", "Arroz con Leche",
        "Empanadillas de Atún", "Paella de Verduras", "Tallarines con Marisco", "Pizza Barbacoa", "Costillas BBQ"
    ]

    # Número aleatorio de pedidos (6–10)
    num_pedidos = random.randint(6, 10)
    print(f"Se generarán {num_pedidos} pedidos aleatorios.\n")

    # Crear la lista de pedidos
    pedidos = []
    for i in range(1, num_pedidos + 1):
        nombre_plato = random.choice(platos_disponibles)
        pedidos.append(Pedido(i, nombre_plato))

    # Crear un lock compartido entre todos los hilos
    lock = threading.Lock()

    # Número aleatorio de cocineros (3–5)
    num_cocineros = random.randint(3, 5)
    print(f" Se asignarán {num_cocineros} cocineros a la cocina.\n")

    # Crear y lanzar cocineros
    cocineros = []
    for i in range(1, num_cocineros + 1):
        cocinero = Cocinero(f"Cocinero {i}", pedidos, lock)
        cocinero.start()
        cocineros.append(cocinero)

    # Esperar a que todos terminen
    for cocinero in cocineros:
        cocinero.join()

    print("\nTodos los pedidos han sido procesados correctamente.")

if __name__ == "__main__":
    main()

File: java/Cocina.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 * Clase principal que lanza la simulación concurrente de la cocina.
 * Genera aleatoriamente pedidos y cocineros, y coordina su ejecución.
 */
public class Cocina {
    public static void main(String[] args) {

        Random random = new Random();

        // Lista de platos disponibles (25 platos distintos)
        String[] platosDisponibles = {
            "Pizza Margarita", "Lasaña Boloñesa", "Sushi Variado", "Paella Mixta", "Tacos de Pollo",
            "Ensalada César", "Hamburguesa Doble", "Canelones de Espinacas", "Sopa de Tomate", "Pollo al Curry",
            "Burritos de Ternera", "Filete con Patatas", "Gazpacho Andaluz", "Pasta Carbonara", "Raviolis de Queso",
            "Huevos Rotos con Jamón", "Croquetas Caseras", "Pulpo a la Gallega", "Tarta de Queso", "Arroz con Leche",
            "Empanadillas de Atún", "Paella de Verduras", "Tallarines con Marisco", "Pizza Barbacoa", "Costillas BBQ"
        };

        // Número aleatorio de pedidos (6–10)
        int numPedidos = random.nextInt(5) + 6;
        System.out.println("Se generarán " + numPedidos + " pedidos aleatorios.\n");

        // Crear la lista de pedidos
        List<Pedido> pedidos = new ArrayList<>();
        for (int i = 1; i <= numPedidos; i++) {
            String nombrePlato = platosDisponibles[random.nextInt(platosDisponibles.length)];
            pedidos.add(new Pedido(i, nombrePlato));
        }

        // Crear el candado compartido
        Object lock = new Object();

        // Número aleatorio de cocineros (3–5)
        int numCocineros = random.nextInt(3) + 3;
        System.out.println("Se asignarán " + numCocineros + " cocineros a la cocina.\n");

        
        // Crear y lanzar todos los cocineros
        // Se usa una lista para poder luego hacer join() y esperar a que terminen.
        List<Cocinero> cocineros = new ArrayList<>();

        for (int i = 1; i <= numCocineros; i++) {
            Cocinero cocinero = new Cocinero("Cocinero " + i, pedidos, lock);
            cocinero.start();        // Se lanza el hilo inmediatamente (flujo concurrente)
            cocineros.add(cocinero); // Se guarda la referencia para el join posterior
        }
       
        
        // Esperar a que todos los cocineros terminen su trabajo
        // join() bloquea el hilo principal hasta que cada hilo finaliza.
        for (Cocinero cocinero : cocineros) {
            try {
                cocinero.join();
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                System.err.println("Error esperando al cocinero " + cocinero.getName());
            }
        }

        System.out.println("\nTodos los pedidos han sido procesados.");
    }
}